flink-core/src/main/java/org/apache/flink/api/common/io/DelimitedInputFormat.java

```diff
@@ -738,15 +739,18 @@ public void reopen(FileInputSplit split, Long state) throws IOException {
 		Preconditions.checkArgument(state == -1 || state >= split.getStart(),
 			" Illegal offset "+ state +", smaller than the splits start=" + split.getStart());
 
-		try {
+		// If we checkpointed at the beginning of split simply call open
+		if (split.getStart() == state) {
 			this.open(split);
-		} finally {
-			this.offset = state;
+			return;
 		}
 
-		if (state > this.splitStart + split.getLength()) {
+		super.open(split);
```

Comment:
what is the main difference between `this.open()` and `super.open()`?
maybe we should give this logic a name and call it explicitly here

Reply:
this.open basically assumes we are at the beginning of the split and immediately fills the readbuffer and tries to drop partial lines at the beginning of the split. It also calls super.open

This behaviour would lead to problems in case we restore from a checkpoint when we never have to do this and seeking to the checkpoint offset can actually mean backward seeking due to this.open's behaviour (and that doesn't work for compressed streams)

Reply:
In general, I understand the logic. I'm just wondering whether we can make the code a bit more maintainable here. Calling `super.open` in `reopen` doesn't look right. Maybe we can just introduce a `openInternal()` with a flag `doSeek` or something similar. In any case we should add more comments here.